### ISSUE TITLE
offset is out of range

### ISSUE THREAD

**Issue by staltz** [flumedb/async-flumelog#11]
We got a crash in production (Manyverse):

```
The value of \\\"offset\\\" is out of range. It must be >= 0 and <= 65534. Received 74465
RangeError [ERR_OUT_OF_RANGE]: The value of \\\"offset\\\" is out of range. It must be >= 0 and <= 65534. Received 74465
    at boundsError (internal/buffer.js:81:9)
    at Buffer.readUInt16LE (internal/buffer.js:238:5)
    at Object.getDataNextOffset (/data/data/se.manyver/files/nodejs-project/index.js:57782:31)
    at Stream._handleBlock (/data/data/se.manyver/files/nodejs-project/index.js:39782:32)
    at Stream._resume (/data/data/se.manyver/files/nodejs-project/index.js:39844:14)
    at Request._callback (/data/data/se.manyver/files/nodejs-project/index.js:57746:9)
    at random-access-storage Request.callback (/data/data/se.manyver/files/nodejs-project/index.js:41966:8)
    at random-access-file onread (/data/data/se.manyver/files/nodejs-project/index.js:68823:31)
    at FSReqCallback.wrapper [as oncomplete] (fs.js:520:5)
```

@arj03

**Comment by arj03** [comment]
Oh that is a very nasty bug. The file got corrupted somehow. I'm guessing this is while migrating, so it did not finish? I guess once way to try and reproduce this would be to kill the migrate process while it is running and continue and see if we can get it to error. After that we can use the bipf file to see what the problem is.

**Comment by Powersource** [comment]
This is on my phone fwiw. App consistently crashes after being open for about 20 seconds.

**Comment by staltz** [comment]
I'm okay doing a new release of Manyverse as soon as we figure out a solution for this bug.

**Comment by arj03** [comment]
Happy coincidence, is it possible to extract the log.bipf from your phone @Powersource ?

**Comment by Powersource** [comment]
Idk really. My phone isn't rooted so gonna guess no? But André probably knows this better. But yeah just let me know how I can help.

**Comment by staltz** [comment]
If it's not rooted, I don't think we can get it out, not without making a new release. @Powersource which dist are you using `-fdroid`, `-indie` or `-googlePlay`?

**Comment by Powersource** [comment]
Indie

Jacob

**Comment by staltz** [comment]
That's better. I may be able to send you some APKs and you can test, until it works, then we make an official release.

**Comment by arj03** [comment]
Been debugging this, some initial observations (debug statements I added):

```
opened file, since: 432506878
failed to decode at 329715493
failed to decode at 329745692
wrong length at end 329774964
```

The block that is wrong, in that it goes over the end, also has a message in there which is not valid bipf, and it is not the last message though.

I've tried migrating my log here using the migrate process, killing the process at random times to see if I could reproduce the problem, no error so far.

I'll keep looking, just wanted to share what I found out so far.
